src/llm/openai_client.py

```diff
@@ -6,8 +6,10 @@ class OpenAI:
     def __init__(self):
         config = Config()
         api_key = config.get_openai_api_key()
+        base_url = config.get_openai_base_url()
         self.client = OAI(
             api_key=api_key,
+            base_url=base_url
```

Comment:
**Security Issue**: Hardcoding or storing the OpenAI API base URL in the application's configuration file without encryption exposes the application to risks if the URL is tampered with. An attacker could redirect the API requests to a malicious endpoint, leading to data leakage or unauthorized actions. <br> **Fix**: Encrypt the OpenAI API base URL in the configuration file and decrypt it when loading the configuration. This prevents unauthorized users from tampering with the API endpoint URL. <br> **Code Suggestion**: 
 ```
 from cryptography.fernet import Fernet

 key = Fernet.generate_key()
 f = Fernet(key)
 encrypted_base_url = f.encrypt(config['API_ENDPOINTS']['OPENAI_BASE_URL'].encode())
 config['API_ENDPOINTS']['OPENAI_BASE_URL'] = encrypted_base_url
 ```